test/model/message_list_test.dart

```diff
@@ -0,0 +1,241 @@
+
+  final store = await TestZulipBinding.instance.globalStore.perAccount(eg.selfAccount.id);
+  store.addUser(eg.user(userId: userId));
+  store.addStream(stream);
+
+  return store;
+}
+
+Future<MessageListView> messageListViewWithMessages(List<Message> messages, PerAccountStore store, Narrow narrow) async {
+  final messageList = MessageListView.init(store: store, narrow: narrow);
+
+  final connection = store.connection as FakeApiConnection;
+
+  connection.prepare(json: GetMessagesResult(
+    anchor: messages.first.id,
+    foundNewest: true,
+    foundOldest: true,
+    foundAnchor: true,
+    historyLimited: false,
+    messages: messages,
+  ).toJson());
+
+  await messageList.fetch();
+
+  check(messageList.messages.length).equals(messages.length);
+
+  return messageList;
+}
+
+void main() async {
+  TestZulipBinding.ensureInitialized();
+
+  const narrow = StreamNarrow(streamId);
+  final stream = eg.stream(streamId: streamId);
+
+  group('update message tests', () {
+
+    test('find message in message list returns index of message', () async {
+      final store = await setupStore(stream);
+
+      final m1 = eg.streamMessage(id: 2, stream: stream);
+      final m2 = eg.streamMessage(id: 4, stream: stream);
+      final m3 = eg.streamMessage(id: 6, stream: stream);
+
+      final messageList = await messageListViewWithMessages([m1, m2, m3], store, narrow);
+      //The implementation of this uses a binary search, so let's test it
+      //a bit more exhaustively.
+
+      //Before the first element
+      var idx = messageList.findMessageWithId(1);
+      check(idx).equals(-1);
+
+      //Is the first element
+      idx = messageList.findMessageWithId(2);
+      check(idx).equals(0);
+
+      //Between first and second
+      idx = messageList.findMessageWithId(3);
+      check(idx).equals(-1);
+
+      //Is the second
+      idx = messageList.findMessageWithId(4);
+      check(idx).equals(1);
+
+      //Between second and third
+      idx = messageList.findMessageWithId(5);
+      check(idx).equals(-1);
+
+      //Is the third
+      idx = messageList.findMessageWithId(6);
+      check(idx).equals(2);
+
+      //After the third
+      idx = messageList.findMessageWithId(7);
+      check(idx).equals(-1);
+
+      //Invalid IDs
+      idx = messageList.findMessageWithId(-8409);
+      check(idx).equals(-1);
+
+      idx = messageList.findMessageWithId(0);
+      check(idx).equals(-1);
+    });
+
+    test('update events are correctly applied to message when it is in the stream', () async {
+      final store = await setupStore(stream);
+
+      const oldContent = "<p>Hello, world</p>";
+      const newContent = "<p>Hello, edited</p>";
+      const newTimestamp = 99999;
+
+      List<String> oldFlags = [];
+      List<String> newFlags = ["starred"];
+
+      final mockMessage = eg.streamMessage(id: 243, stream: stream, content: oldContent, flags: oldFlags);
```

Comment:
This isn't something I'd call a "mock message" — it's a perfectly genuine `Message` object. (I believe all the data in it is something one totally could elicit from a real Zulip server, even; and if there's some aspect that makes that not true, then that's something we should be able to fix in the `eg` / `example_data.dart` code.)

One good name would be simply `message`. In at least some of these test cases, the later variable called `message` could be instead `actual` or `actualMessage`, or could just be inlined in the one place it's used.

Could also call this something like `inputMessage` or `origMessage` to distinguish it from a later `message`.